### DISCUSSION TITLE
Offline counterpart to (interactive) BSP

### DISCUSSION THREAD

**Discussion by bts** [build-server-protocol/build-server-protocol#704] [Proposals]
Hi there,

Thanks for this project. This is very exciting. One of the best things about LSP is that there's a batch/offline counterpart in LSIF (Language Server Index Format). LSIF enables clients to run massive, high throughput analyses over *all* source information at once in a consistent manner. No overhead from a long series of RPC calls.

The one shortcoming of LSIF though, IIUC, is that if one file in the repo changes, everything (or a lot?) needs to be recomputed due to the way that IDs are managed. SCIP from sourcegraph improves on LSIF by producing batch indices that are amenable to incremental updates.

Has there ever been any discussion about creating an analogue to SCIP for BSP?

Thanks!
Brian

**Comment by jastice** [comment]
Interesting thought. There hasn't been any discussion on the topic, but it sounds interesting. What kind of use cases do you imagine?

**Comment by bts** [comment]
The ability to run any kind of dependency, reachability, etc. analysis over build graphs without having to implement it separately for each different build system. e.g. As part of CI or some offline analytic job, I might want to do some processing for targets in a build graph that were affected by a commit. Or I might want to leverage (e.g. bazel) service definitions to help determine which code can affect other code across service boundaries.
